easybuild/easyblocks/s/superlu.py

```diff
@@ -0,0 +1,148 @@
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+from easybuild.tools.modules import get_software_root,get_software_version
+
+
+class EB_SuperLU(CMakeMake):
+    """
+    Support for building the SuperLU library
+    """
+
+    @staticmethod
+    def extra_options():
+        """
+        Define custom easyconfig parameters for SuperLU.
+        """
+        extra_vars = {
+            'build_shared_libs': [False, "Build shared library (instead of static library)", CUSTOM],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """
+        Set the CMake options for SuperLU
+        """
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['build_shared_libs']:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=ON')
+        else:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=OFF')
+
+        # Add -fPIC flag if necessary
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=ON')
+        else:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=OFF')
+
+        # Make sure not to build the slow BLAS library included in the package
+        self.cfg.update('configopts', '-Denable_blaslib=OFF')
+
+        # Set the BLAS library to use
+        # For this, use the BLA_VENDOR option from the FindBLAS module of CMake
+        # Check for all possible values at https://cmake.org/cmake/help/latest/module/FindBLAS.html
+        toolchain_blas = self.toolchain.definition().get('BLAS', ['NO_BLAS_IN_TOOLCHAIN'])[0]
+        if toolchain_blas == 'imkl':
+            imkl_version = get_software_version('imkl')
+            if LooseVersion(imkl_version) >= LooseVersion('10'):
+                # 'Intel10_64lp' -> For Intel mkl v10 64 bit,lp thread model, lp64 model
+                # It should work for Intel MKL 10 and above, as long as the library names stay the same
+                # SuperLU requires thread, 'Intel10_64lp_seq' will not work!
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel10_64lp"')
+            else:
+                # 'Intel' -> For older versions of mkl 32 and 64 bit
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel"')
+        elif toolchain_blas == 'ACML':
+            self.cfg.update('configopts', '-DBLA_VENDOR="ACML"')
+        elif toolchain_blas == 'ATLAS':
+            self.cfg.update('configopts', '-DBLA_VENDOR="ATLAS"')
+        elif toolchain_blas == 'OpenBLAS':
+            # Unfortunately, OpenBLAS is not recognized by FindBLAS from CMake,
+            # we have to specify the OpenBLAS library manually
+            self.cfg.update('configopts', '-DBLAS_LIBRARIES="${EBROOTOPENBLAS}/lib/libopenblas.a;-pthread"')
```

Comment:
use `get_software_libdir('OpenBLAS')` instead of `${EBROOTOPENBLAS}/lib`? That's clearer (full path specified), and it deals with the possibility that the library is actually in `lib64` (for whatever reason).